htdocs/src/Oc/Page/PageRepository.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Oc\Page;
+
+use Doctrine\DBAL\Connection;
+use InvalidArgumentException;
+
+/**
+ * Class PageRepository
+ *
+ * @package Oc\Page
+ * @author Nick Lubisch <oc@nicklubisch.de>
+ */
+class PageRepository
+{
+    /**
+     * Database table name that this repository maintains.
+     *
+     * @var string
+     */
+    const TABLE = 'page';
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * PageRepository constructor.
+     *
+     * @param Connection $connection
+     */
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * Fetches a page by slug.
+     *
+     * @param array $where
+     *
+     * @return null|PageEntity
+     */
+    public function fetchOneBy(array $where = [])
+    {
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->select('*')
+            ->from(self::TABLE)
+            ->setMaxResults(1);
+
+
+        if (count($where) > 0) {
+            foreach ($where as $column => $value) {
+                $queryBuilder->andWhere($column . ' = ' .  $queryBuilder->createNamedParameter($value));
+            }
+        }
+
+        $statement = $queryBuilder->execute();
+
+        $result = $statement->fetch();
+
+        if ($result === false) {
+            return null;
```

Comment:
should we throw an Exception for this case?

Reply:
The new discussed behaviour, that the repository throws exceptions like RecordNotFoundException, RecordsNotFoundException, RecordNotPersistedException and RecordAlreadyExistsException and the service handels them to return default values like an empty array or null.